projects/http-auth/main.go

```diff
@@ -0,0 +1,88 @@
+package main
+
+import (
+	"fmt"
+	"html"
+	"io"
+	"log"
+	"net/http"
+	"os"
+
+	"github.com/joho/godotenv"
+	"golang.org/x/time/rate"
+)
+
+func main() {
+	err := godotenv.Load()
+	if err != nil {
+		log.Fatal("Error loading .env file")
+	}
+	limiter := rate.NewLimiter(100, 30)
+	if limiter.Allow() {
+		http.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
```

Comment:
So, I think you have a little misconception here about what this is doing.
What line 22 - the call to http.HandleFunc - is doing is telling the HTTP server how to handle it when someone requests the '/' path.
Whenever this happens, the http server will call the func which you are defining between lines 23-50.
So if you want to rate limit that, then the call to the limiter should be done INSIDE your handler - because as your code is now, limiter.Allow will only happen once, which doesn't really make sense.